### PR TITLE
Refactoring of build (fix #135)

### LINKED ISSUE #135: refactor project_builder
separate `build` method into 2:
- [ ] `build_srpm(spec_file, tarball, srpm_output_dir)` - all params are Pathlib objects, rpmbuild call
- [ ] `build_rpm(srpm_path, distro, arch)` - TODO mock call, add just method definition + existing code

use Command class helper

### PR DESCRIPTION
**Pull request by regeciovad**

### CODE REVIEW

**Comment by jsilhan** [comment]
add test, please.
- [ ] create spec for hello package
- [ ] initialize Bae object
- [ ] base.package_builde.build_srpm()
- [ ] test if exists base.srpm_path()

**Comment by jsilhan** [comment]
We need to create FakeBase inheriting from Base and implement empty `__init__`, and setters [1] for `self.project_name` and `self.base_dir`. Also make build_srpm static.

**Comment by regeciovad** [comment]
The test creates a src.rpm package and both (test and build_srpm method) look fine.

File: rpg/package_builder.py [hunk 1]
```diff
@@ -21,16 +22,21 @@ def _parse_error(self, log_path):
...
+    def build_srpm(spec_file, tarball, srpm_output_path):
+        p = PackageBuilder()
+        call(["rpmdev-setuptree", ""])
+        user_home = expanduser("~")
+        copy_file(str(spec_file), user_home + "/rpmbuild/SPECS/")
+        copy_file(str(tarball), user_home + "/rpmbuild/SOURCES/")
+        output = Command("rpmbuild -bs " + user_home + "/rpmbuild/SPECS/" 
+            + spec_file.name).execute()
+        srpm_path = p._get_last_word(output)
+        copy_file(str(srpm_path), str(srpm_output_path))
```
Comment (jsilhan) [inline rpg/package_builder.py]: it should move it. We don;t want extra copy

File: rpg/package_builder.py [hunk 2]
```diff
@@ -42,3 +48,9 @@ def build(self, spec_file, tarball, distro=None, arch=None):
         result = str(self._get_last_word(logs).replace("\\n'", ""))
         if not "Finish: run" in line:
             return self._parseError(result)
+    
+    def build(self, spec_file, tarball, distro=None, arch=None):
```
Comment (jsilhan) [inline rpg/package_builder.py]: comment all lines in this method and add there `# FIXME completely broken` - don't fix it now, it could be done later.

**Comment by jsilhan** [comment]
Fixup the commits as much as possible. It should have been 1  or maybe 3 commits. You should have not fix the feature, that was not pushed yet, in separate commit. See "Small changes in build_srpm". Good job, just clean it and update it, please.

File: tests/test_build_srpm.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+class BuildSrpmTest(RpgTestCase):
+    
+    def test_build_srpm(self):
+        fbase = FakeBase()
+        PackageBuilder.build_srpm(fbase.spec_path, fbase.archive_path, fbase.base_dir)
+        self.assertTrue(fbase.srpm_path.exists())
+        
+    def TearDown():
+        base = Base()
```
Comment (jsilhan) [inline tests/test_build_srpm.py]: Didn't you mean `FakeBase()`?

**Comment by jsilhan** [comment]
WRT commits it's perfect although the tests didn't pass:
```
$ nosetests-3.4 tests/
..E..............F.......
======================================================================
ERROR: test_build_srpm (test_build_srpm.BuildSrpmTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jsilhan/Documents/rpg/tests/test_build_srpm.py", line 24, in test_build_srpm
    self.assertTrue(fbase.srpm_path.exists())
AttributeError: 'FakeBase' object has no attribute 'srpm_path'

======================================================================
FAIL: test_find_files (test_plugins.FindPatchPluginTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jsilhan/Documents/rpg/tests/test_plugins.py", line 50, in test_find_files
    sorted_files)
AssertionError: Lists differ: [('/M[54 chars], ('/hello_project/hello-1.4-1.fc21.src.rpm', [417 chars]one)] != [('/M[54 chars], ('/libs/libdynamic.so.1', None, None), ('/li[268 chars]one)]

First differing element 2:
('/hello_project/hello-1.4-1.fc21.src.rpm', None, None)
('/libs/libdynamic.so.1', None, None)

First list contains 3 additional elements.
First extra element 11:
('/py/requires/sourcecode2.py', None, None)

  [('/Makefile', None, None),
   ('/c/sourcecode.c', None, None),
-  ('/hello_project/hello-1.4-1.fc21.src.rpm', None, None),
-  ('/hello_project/hello-1.4.tar.gz', None, None),
-  ('/hello_project/hello.spec', None, None),
   ('/libs/libdynamic.so.1', None, None),
   ('/libs/libstatic.a', None, None),
   ('/patch/0.patch', None, None),
   ('/patch/1.patch', None, None),
   ('/patch/2.patch', None, None),
   ('/py/plugin0.py', None, None),
   ('/py/requires/sourcecode2.py', None, None),
   ('/py/sourcecode.py', None, None),
   ('/translation/CZ.mo', None, None)]

----------------------------------------------------------------------
Ran 25 tests in 0.188s

FAILED (errors=1, failures=1)
```

File: tests/test_build_srpm.py (hunk 3 above)
Comment (regeciovad) [inline tests/test_build_srpm.py]: I don't understand - I am supposed to join BuildSrpmTest and FakeBase or it's about inheriting?

**Comment by regeciovad** [comment]
Yes, this is not error in test. I "deleted" merge from master to my branch and there were things I was supposed to test it. But with merge branch it should be running ok.
Can I do something with that?

**Comment by jsilhan** [comment]
Right, it wasn't rebased so I have seen the failures. This is what I had in mind 348c475 . You were deleting the wrong directory Base.base_dir x FakeBase.base_dir. Merged, thanks.

**Pull request closed without merging**